org.adempiere.base.process/src/org/compiere/process/M_PriceList_Create.java

```diff
@@ -117,10 +117,7 @@ private String checkPrerequisites() {
 				"UPDATE M_Product_PO SET PriceLastPO = 0 WHERE PriceLastPO IS NULL AND AD_Client_ID=?",
 				clientParam, get_TrxName());
 		DB.executeUpdateEx(
-				"UPDATE M_Product_PO SET PricePO = PriceLastPO WHERE (PricePO IS NULL OR PricePO = 0) AND PriceLastPO <> 0 AND AD_Client_ID=?",
-				clientParam, get_TrxName());
-		DB.executeUpdateEx(
-				"UPDATE	M_Product_PO SET PricePO = 0 WHERE PricePO IS NULL AND AD_Client_ID=?",
+				"UPDATE M_Product_PO SET PricePO = PriceLastPO WHERE PricePO IS NULL OR (PricePO = 0 AND PriceLastPO <> 0) AND AD_Client_ID=?",

```

Comment:
what does it fix here ?

Reply:
Hi,
not a fix. Only a small enhancement because it replaces the previously following update.

Reply:
Doesn't sound like an enhancement.

The change in the WHERE sounds wrong:
was:
(PricePO IS NULL OR PricePO = 0) AND PriceLastPO <> 0 AND AD_Client_ID=?
changed to:
PricePO IS NULL OR (PricePO = 0 AND PriceLastPO <> 0) AND AD_Client_ID=?

That's a very different set of records, I have not tested, but I have the feeling it breaks some scenarios.

I think is better not to mix features in the middle of another ticket, if a small enhancement is possible, then better to discuss it.